frontend/src/control/fader.rs

```diff
@@ -142,6 +144,16 @@ impl Component for Fader {
                 self.mouse_mode = MouseMode::Normal;
                 true
             }
+            FaderMsg::Scroll(dir) => {
+                let delta = match dir {
+                    Scroll::Up(delta) => delta,
+                    Scroll::Down(delta) => delta * -1.0,
+                };
+                let factor = 0.0001;
+                let fader_value = util::clamp(0.0, 1.0, self.props.value + delta * factor);
```

Comment:
There is definitely a race condition of sorts here. On really fast scrolls, this `self.props.value` is likely stale. At least, that would explain some jitter on the fader UI (bounces up and down while scrolling aggressively in a single direction). I think I haven't internalised this data flow well enough to fully reason through this though, yet.

Reply:
Wonder if setting `self.props.value` to the new `fader_value` would sort it out? Or if we'd like to keep props pristine, keeping some internal current value state which is updated on props change

Reply:
Hmm I think that would mess with the uni-directional data flow and possibly mean the parent component wouldn't re-render. This would definitely be problematic in React (and impossible in Elm). It might be incidentally possible here, but I'd want to think about the implications more closely first.

Reply:
The props is owned by the component - only by convention/convenience do we even keep the actual struct around.

Actually, I'm thinking something like this could work:

```rust
struct ScrollTarget {
    props: Props,
    value: f64,
}

struct Props {
    value: f64,
}

impl Component for ScrollTarget {
    fn change(&mut self, new_props: Props) -> ShouldRender {
        if new_props.value != self.props.value {
            self.value = new_props.value;
            true
        } else {
            false
        }
    }
}
```

Then we consistently use `self.value` rather than `self.props.value` in the component. We can even update `self.value` so we have a self-consistent view of what the value is when it's changed before the props actually change

Reply:
Except we don't have a `value` in `ScrollTarget` (or indeed the `wheel`/`scroll` event), just a delta. It's not the `ScrollTarget`'s place to know how that delta would be applied without it being coupled to specific use cases.

Or do you mean:

``` rust
// in point of use (e.g. fader.rs)
html! {
	<ScrollTarget
        value={self.props.value} // initialise with a fader
    >
      // ... 
    </ScrollTarget>
}
```

If that direction, I think it would need to be a callback of some kind because ultimately the parent component needs to react to changes in that value.

What am I missing?

Reply:
To replicate this jitteriness, scroll very aggressively in one direction or another.

Here's me doing it with mouse whose physical mouse button has inertia and so remains spinning as I move the mouse around different targets:
 
![screen](https://user-images.githubusercontent.com/2560/78849462-e753e580-7a57-11ea-9f7d-126d1726d9d8.gif)

The fps of that gif is only 15 so it's actually making it look less severe than it really is.

I've noticed if the backend is killed, the jitteriness stops. So I'm pretty sure this is re-renders of the component fighting new values coming in. The `value + delta` result which is emitted is already out of date so resetting the fader.